packages/passport/sdk/src/utils/metrics.ts

```diff
@@ -0,0 +1,39 @@
+import { trackError, trackFlow } from '@imtbl/metrics';
+
+export const withMetrics = <T>(
+  fn: () => T,
+  event: string,
+): T => {
+  const flow = trackFlow('passport', event);
+
+  try {
+    return fn();
```

Comment:
LGTM 👍 
We may want to extend these implementations in the future to pass the flow to the function being executed, e.g:

```suggestion
  fn: (flow: Flow) => T,
  event: string,
): T => {
  const flow = trackFlow('passport', event);

  try {
    return fn(flow);
```